src/app/login/page.tsx

```diff
@@ -5,6 +5,9 @@ import Image from "next/image";
 import Paragraph from "../components/Paragraph";
 import SpotifyLogo from "../../../public/images/spotify_logo.png"
 import SpotifyLogo2 from "../../../public/images/Spotify_Logo_2.png"
+import Link from "next/link";
+
+const AUTH_URL = "https://accounts.spotify.com/en/authorize?response_type=token&client_id=a96eef12a4ff426fa590bc684129730c&scope=ugc-image-upload%20playlist-modify-private%20playlist-modify-public%20streaming%20user-read-email%20user-top-read&redirect_uri=http://localhost:3000/personalised-homepage"
```

Comment:
Here we're using `localhost` as the `redirect_uri`. This would only work if the user had a local server running our application.

Does this accept relative paths [i.e. only having `/personalised-homepage`]? If not, we might want to change this to the Vercel domain.

Reply:
I will investigate relative paths option here. The redirect URL is also stored in the spotify web developers app, I believe they state HTTP/ must be part of the redirect_url there.